service/matching/handler/engine.go

```diff
@@ -162,6 +163,7 @@ func NewEngine(
 }
 
 func (e *matchingEngineImpl) Start() {
+	e.registerDomainFailoverCallback()
```

Comment:
Matching engine is not created on-demand so it doesn't matter probably but for consistency reasons let's unregister during Stop.

Reply:
The matching engine can be completely removed and everything can be put inside handler directly, but the change is not unnecessary.